src/components/modals/settings-modal/settings-modal.tsx

```diff
@@ -0,0 +1,71 @@
+import { Modal, Select } from "antd";
+import { useState } from "react";
+import "./settings-modal.css";
+import { useTheme } from "../../utilities/useTheme";
+import { useTextDirection } from "../../utilities/useTextDirection";
+import { changeLanguage } from "i18next";
+
+type SettingsModalProps = {
+  visible: boolean;
+  onOk: () => void;
+  onCancel: () => void;
+};
+
+export const SettingsModal = ({
+  visible,
+  onOk,
+  onCancel,
+}: SettingsModalProps) => {
+  const [language, setLanguage] = useState("en");
+  const [currentTheme, , toggleTheme] = useTheme();
+
+  const changeDirWithLanguage = useTextDirection();
+
+  const handleThemeChange = (value: string) => {
+    if (value === "dark" && currentTheme === "light") {
+      toggleTheme();
+    } else if (value === "light" && currentTheme === "dark") {
+      toggleTheme();
+    }
+  };
+
+  const handleLanguageChange = (value: string) => {
+    setLanguage(value);
+    changeLanguage(value);
+    changeDirWithLanguage(value);
+  };
+
+  return (
+    <Modal
+      title="Settings"
+      open={visible}
+      onOk={onOk}
+      onCancel={onCancel}
+      maskClosable={false}
+    >
+      <div className="settings-option">
+        <label>Language:</label>
+        <Select
+          className="settings-select"
+          value={language}
+          onChange={handleLanguageChange}
+        >
+          <Select.Option value="en">English</Select.Option>
+          <Select.Option value="he">עברית</Select.Option>
+        </Select>
+      </div>
+
+      <div className="settings-option">
+        <label>Theme:</label>
```

Comment:
There is already  a theme switcher component [https://github.com/ofir-zeitoun/expenses-client/blob/main/src/components/theme-switcher/theme-switcher.tsx](Theme Switcher )

Reply:
As Ofir requested  [here](https://github.com/ofir-zeitoun/expenses-client/issues/47#issuecomment-2395474073) 
i added alternative way to toggle between theme